number_parser/parser.py

```diff
@@ -252,20 +259,37 @@ def parse_number(input_string, language=None):
     if language is None:
         language = _valid_tokens_by_language(input_string)
 
-    lang_data = LanguageData(language)
+    if numeral_systems is None:
+        numeral_systems = NUMERAL_SYSTEMS
+
+    if _search_roman(input_string):
+        numeral_systems = ['roman']
+
+    else:
+        numeral_systems = ['decimal']
+
+    for numeral_system in numeral_systems:
+        if numeral_system == 'decimal':
+            lang_data = LanguageData(language)
+
+            tokens = _tokenize(input_string, language)
+            normalized_tokens = _normalize_tokens(tokens)
+            for index, token in enumerate(normalized_tokens):
+                if _is_cardinal_token(token, lang_data) or not token.strip():
+                    continue
+                if _is_skip_token(token, lang_data) and index != 0:
+                    continue
+                return None
+            number_built = _build_number(normalized_tokens, lang_data)
+            if len(number_built) == 1:
+                return int(number_built[0])
+            return None
 
-    tokens = _tokenize(input_string, language)
-    normalized_tokens = _normalize_tokens(tokens)
-    for index, token in enumerate(normalized_tokens):
-        if _is_cardinal_token(token, lang_data) or not token.strip():
-            continue
-        if _is_skip_token(token, lang_data) and index != 0:
-            continue
-        return None
-    number_built = _build_number(normalized_tokens, lang_data)
-    if len(number_built) == 1:
-        return int(number_built[0])
-    return None
+        elif numeral_system == 'roman':
+            return int(_parse_roman(input_string))
+
+        else:
+            raise ValueError(f'"{numeral_system}" is not a supported numeral system')
```

Comment:
`!r` is a nice trick to get the value to be represented however it would be in Python (e.g. strings are quoted). It’s slightly better because if the input was of an invalid type (e.g. `numeral_systems=[b'roman']`) this would make it obvious, as it would print b'roman' instead of "roman" (or maybe there would be an exception because of the bytes needing encoding, come to think of it; not sure).

```suggestion
            raise ValueError(f'{numeral_system!r} is not a supported numeral system')
```